judge/views/two_factor.py

```diff
@@ -51,9 +51,15 @@ class TOTPEnableView(TOTPView):
 
     def get(self, request, *args, **kwargs):
         profile = self.profile
+        update_fields = []
         if not profile.totp_key:
             profile.totp_key = pyotp.random_base32(length=32)
-            profile.save()
+            update_fields.append("totp_key")
+        if not profile.scratch_codes:
+            profile.scratch_codes = json.dumps([pyotp.random_base32(length=16)
```

Comment:
I don't particularly like this duplicating of code for scratch code generation. I think either `generate_scratch_codes` should be modified to not call `save()`, or inline `generate_scratch_codes` in the one place that it is called (since right now, there's a dedicated method to generating scratch codes for that one place, but then it's inlined here).